core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/UnifiedResourceSchema.scala

```diff
@@ -0,0 +1,114 @@
+      fileSize: Field[UInteger] = DSL.inline(null, classOf[UInteger]),
+      fileUserAccess: Field[UserFileAccessPrivilege] =
+        DSL.inline(null, classOf[UserFileAccessPrivilege])
+  ): UnifiedResourceSchema = {
+    new UnifiedResourceSchema(
+      Seq(
+        resourceType -> resourceType.as("resourceType"),
+        name -> name.as("resource_name"),
+        description -> description.as("resource_description"),
+        creationTime -> creationTime.as("creation_time"),
+        wid -> wid.as("wid"),
+        workflowLastModifiedTime -> workflowLastModifiedTime.as("last_modified_time"),
+        workflowUserAccess -> workflowUserAccess.as("workflow_privilege"),
+        projectsOfWorkflow -> projectsOfWorkflow.as("projects"),
+        uid -> uid.as("uid"),
+        userName -> userName.as("userName"),
+        userEmail -> userEmail.as("email"),
+        pid -> pid.as("pid"),
+        projectOwnerId -> projectOwnerId.as("owner_uid"),
+        projectColor -> projectColor.as("color"),
+        fid -> fid.as("fid"),
+        fileOwnerId -> fileOwnerId.as("owner_id"),
+        fileUploadTime -> fileUploadTime.as("upload_time"),
+        filePath -> filePath.as("path"),
+        fileSize -> fileSize.as("size"),
+        fileUserAccess -> fileUserAccess.as("user_file_access")
+      )
+    )
+  }
+}
+
+/**
+  * Refer to texera/core/scripts/sql/texera_ddl.sql to understand what each attribute is
+  *
+  * Common Attributes (4 columns): All types of resources have these 4 attributes
+  * 1. `resourceType`: Represents the type of resource (`String`). Allowed value: project, workflow, file
+  * 2. `name`: Specifies the name of the resource (`String`).
+  * 3. `description`: Provides a description of the resource (`String`).
+  * 4. `creation_time`: Indicates the timestamp of when the resource was created (`Timestamp`). It represents upload_time if the resourceType is `file`
+  *
+  * Workflow Attributes (6 columns): Only workflow will have these 6 attributes.
+  * 5. `WID`: Represents the Workflow ID (`UInteger`).
+  * 6. `lastModifiedTime`: Indicates the timestamp of the last modification made to the workflow (`Timestamp`).
+  * 7. `privilege`: Specifies the privilege associated with the workflow (`Privilege`).
+  * 8. `UID`: Represents the User ID associated with the workflow (`UInteger`).
+  * 9. `userName`: Provides the name of the user associated with the workflow (`String`).
+  * 10. `projects`: The project IDs for the workflow, concatenated as a string (`String`).
+  *
+  * Project Attributes (3 columns): Only project will have these 3 attributes.
+  * 11. `pid`: Represents the Project ID (`UInteger`).
+  * 12. `ownerId`: Indicates the ID of the project owner (`UInteger`).
+  * 13. `color`: Specifies the color associated with the project (`String`).
+  *
+  * File Attributes (7 columns): Only files will have these 7 attributes.
+  * 14. `ownerUID`: Represents the User ID of the file owner (`UInteger`).
```

Comment:
a major comment here about the naming. we actually have duplicated fields from those three resources. for example, each resource has a "user id". right now, you named the "user id" differently: in workflow, it is called "UID"; in project, it is called "ownerId"; in file, it is called "ownerUID". I think it is very confusing and we will get lost in the future. 

One idea may be make this user id as a common field (just like your first 4 columns). Besides user Id, each resource also has "access" information, which can be made as a common field. So you could do 
`common fields (resource type, owner id, access level)` + `resource specific fields`.

Or another way, you at least need to rename them in a consistent way. For example, you could name it to `workflow_owner_id`, `project_owner_id` and `file_owner_id`, despite they are the same information.

Reply:
moved ownerId into the comment fields, as well as the lastModifiedTime.